### PR TITLE
Fixes #2326: Google Earth export GPS glitch filtering and lat/lon syn…

### LINKED ISSUE #2326: Google Earth export
Lat/lon data tends to be out of sync. The export from logs seems to pick samples when there is a change in lat/lon, even if just one of them has changed. However, when the next coordinate get updated in the next sample, this seems to be ignored. 
This leads to stepped tracks. Some filtering would be nice. Like; if just one coordinate has changed, pick the next sample (which seems to contain both, even at 10Hz sampling).

Log file:
http://www.rcgroups.com/forums/showatt.php?attachmentid=7887955&d=1432153024

ge

### PR DESCRIPTION
**Pull request by projectkk2glider**
…chronization

Closes #2326 

Done a couple of related things:
* a little code refactoring
* fixes #2326 
* filters out any GPS glitches and zero lat/lon location
* fixed icon path

### CODE REVIEW

File: companion/src/helpers.h [hunk 1]
```diff
@@ -162,4 +162,31 @@ QSet<QString> getFilesSet(const QString &path, const QStringList &filter, int ma
 
 bool caseInsensitiveLessThan(const QString &s1, const QString &s2);
 
+
+class gpsGlitchFilter
```
Comment (bsongis) [inline companion/src/helpers.h]: First letter should be uppercase!

File: companion/src/helpers.cpp [hunk 2]
```diff
@@ -1097,3 +1097,53 @@ bool caseInsensitiveLessThan(const QString &s1, const QString &s2)
...
+  lastLat = latitude;
+  lastLon = longitude;
+  return true;
+}
+
+double toDecimalCoordinate(QString value) 
+{
+  double temp = int(value.left(value.length()-1).toDouble() / 100);
+  double result = temp + (value.left(value.length() - 1).toDouble() - temp * 100) / 60.0;
+  if ((value.right(1) == "S") || (value.right(1) == "W")) {
```
Comment (bsongis) [inline companion/src/helpers.cpp]: It would be better to compare a char, not a QString!

File: companion/src/helpers.h [hunk 3]
```diff
@@ -162,4 +162,31 @@ QSet<QString> getFilesSet(const QString &path, const QStringList &filter, int ma
...
+  int glitchCount;
+  double lastLat;
+  double lastLon;
+};
+
+class gpsLatLonFilter
+{
+public:
+  gpsLatLonFilter() {};
+  bool isValid(QString latitude, QString longitude);
```
Comment (bsongis) [inline companion/src/helpers.h]: Those args should be const QString & (unless they are modified). It will avoid a couple of memcpy ...

**Comment by bsongis** [comment]
Thanks, the new map is really nice!

File: companion/src/helpers.cpp [hunk 4]
```diff
@@ -1097,3 +1097,55 @@ bool caseInsensitiveLessThan(const QString &s1, const QString &s2)
...
+}
+
+double toDecimalCoordinate(const QString & value) 
+{
+  if (value.isEmpty()) return 0.0;
+  double temp = int(value.left(value.length()-1).toDouble() / 100);
+  double result = temp + (value.left(value.length() - 1).toDouble() - temp * 100) / 60.0;
+  QChar direction = value.at(value.size()-1);
+  if ((direction == 'S') || (direction == 'W')) {
+    result = -result;
```
Comment (projectkk2glider) [inline companion/src/helpers.cpp]: @bsongis is this what you had in mind? I am not sure if it is any faster.

Comment (bsongis) [inline companion/src/helpers.cpp]: Perfect, and yes I really think it's faster!

**Pull request merged**